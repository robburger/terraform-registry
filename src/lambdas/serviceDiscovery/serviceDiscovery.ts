import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda'

import { Message } from '../../utils'

export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
  return Message.success({
    'modules.v1': `https://${event.headers.Host}/modules/v1/`,
    'providers.v1': `https://${event.headers.Host}/providers/v1/`,
  })
}
