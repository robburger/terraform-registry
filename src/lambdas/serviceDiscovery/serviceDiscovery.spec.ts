import { APIGatewayProxyEvent } from 'aws-lambda'
import { handler } from './serviceDiscovery'

import * as data from '../../../test/events/serviceDiscovery/ServiceDiscovery.json'

const mockEvent: APIGatewayProxyEvent = data

describe('serviceDiscovery handler', () => {
  it('should respond with a valid Terraform Service Discovery object', async () => {
    const result = await handler(mockEvent)
    expect(result).toMatchSnapshot()
  })
})
