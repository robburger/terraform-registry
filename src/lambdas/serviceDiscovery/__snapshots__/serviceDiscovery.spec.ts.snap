// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serviceDiscovery handler should respond with a valid Terraform Service Discovery object 1`] = `
Object {
  "body": "{\\"modules.v1\\":\\"https://127.0.0.1:3000/modules/v1/\\",\\"providers.v1\\":\\"https://127.0.0.1:3000/providers/v1/\\"}",
  "headers": Object {
    "content-type": "application/json",
  },
  "statusCode": 200,
}
`;
